tests/dialects/test_dialects.py

```diff
@@ -0,0 +1,18 @@
+from xdsl.dialects import get_all_dialects
+
+
+def test_op_names():
+    """
+    Make sure that all operation class names match our convention of having an "Op"
+    suffix.
+    """
+    all_dialects = get_all_dialects()
+    malformed_op_names = tuple(
+        (op.name, op.__name__)
+        for dialect_factory in all_dialects.values()
+        for op in dialect_factory().operations
+        if "cmath" not in op.name
```

Comment:
It's nicer to check the dialect name rather than check if each op contains "cmath". The above is a little yucky though I couldn't see a nice way of doing it so maybe just leave it as is

Reply:
I agree but I think that might be fixing the wrong problem, I filed this to track the issue: #3527

Reply:
absolutely